fendermint/testing/materializer/src/materializer.rs

```diff
@@ -0,0 +1,246 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+use async_trait::async_trait;
+use ethers::types::H160;
+use fvm_shared::econ::TokenAmount;
+use std::collections::BTreeMap;
+use tendermint_rpc::Url;
+
+use fendermint_vm_genesis::Collateral;
+
+use crate::{
+    manifest::Balance, AccountName, NodeName, RelayerName, ResourceHash, SubnetName, TestnetName,
+};
+
+/// The materializer is a component to provision resources of a testnet, and
+/// to carry out subsequent commands on them, e.g. to restart nodes.
```

Comment:
So basically the materializer implementation provides the concrete implementation of the backing resource (it being docker, local system, raw cli commands, etc) and the Testnet orchestrates these command to get to a valid testnet deployment.

Reply:
Yes, that's pretty much all it does at the moment.